.github/workflows/ci-ptf.yml

```diff
@@ -43,11 +62,28 @@ jobs:
           key: test-${{ env.UNIFIED }}-${{ runner.os }}
           max-size: 1000M
 
+      - name: Cache VM image
+        uses: actions/cache@v3.0.5
+        id: vm_cache
+        with:
+          path: ${{ env.DISK_IMAGE }}
+          key: ${{ env.OS_TYPE }} ${{ env.KERNEL_VERSIONS }}
+
+      - name: Prepare runner
+        run: sudo -E ./tools/ci-ptf/prepare_runner.sh
+
+      - name: Build VM
+        if: steps.vm_cache.outputs.cache-hit != 'true'
+        run: sudo -E ./tools/ci-ptf/build_vm.sh
+
       - name: Build (Linux)
         run: |
-          docker build --network host -t p4c --build-arg MAKEFLAGS=-j8 --build-arg IMAGE_TYPE=test --build-arg ENABLE_UNIFIED_COMPILATION=$UNIFIED --build-arg ENABLE_GMP=$ENABLE_GMP --build-arg INSTALL_PTF_EBPF_DEPENDENCIES=ON .
+          docker build --network host -t p4c --build-arg MAKEFLAGS=-j8 --build-arg IMAGE_TYPE=test --build-arg ENABLE_UNIFIED_COMPILATION=$UNIFIED --build-arg ENABLE_GMP=$ENABLE_GMP --build-arg INSTALL_PTF_EBPF_DEPENDENCIES=ON --build-arg KERNEL_VERSIONS="$KERNEL_VERSIONS" .
           ./tools/export_ccache.sh
 
+      - name: Install VM
+        run: sudo -E ./tools/ci-ptf/install_vm.sh
+
       - name: Run PTF tests for eBPF backend (Linux)
         run: |
-          sudo docker run --privileged -v /sys/fs/bpf:/sys/fs/bpf -w /p4c/backends/ebpf/tests p4c ./test.sh
+          sudo -E ./tools/ci-ptf/run_for_each_kernel.sh sudo docker run --privileged -v /sys/fs/bpf:/sys/fs/bpf -w /p4c/backends/ebpf/tests p4c ./test.sh
```

Comment:
Can we define parametrized `run_test.sh` instead of the bash script that iterates over kernel versions?

Reply:
or parametrized CI jobs